pymc/distributions/multivariate.py

```diff
@@ -2205,20 +2202,17 @@ def _infer_shape(self, size, dist_params, param_shapes=None):
 
         size = tuple(size)
 
-        return size + (K + 1,)
+        return size + tuple(alpha.shape) + (K + 1,)
 
     @classmethod
     def rng_fn(cls, rng, alpha, K, size):
         if K < 0:
             raise ValueError("K needs to be positive.")
 
-        if size is None:
-            size = (K,)
-        elif isinstance(size, int):
-            size = (size,) + (K,)
-        else:
-            size = tuple(size) + (K,)
+        distribution_shape = alpha.shape + (K,)
+        size = to_tuple(size) + distribution_shape
```

Comment:
Again, size should not simply be prepended to the shape of alpha. Size (when specified) must include (but can go beyond) any batched dimensions implied by alpha. This guide might make this point more clear: https://www.pymc.io/projects/docs/en/stable/learn/core_notebooks/dimensionality.html